### PR TITLE
[Data] Add stable diffusion benchmark

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

This PR adds a nightly test that benchmarks stable diffusion batch inference.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/nightly_tests/dataset/stable_diffusion_benchmark.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+from typing import Dict
+
+import numpy as np
+import torch
+from diffusers import StableDiffusionImg2ImgPipeline
+
+import ray
+
+DATA_URI = "s3://air-example-data-2/300G-image-data-synthetic-raw-parquet/"
+BATCH_SIZE = 20
```
Comment (amogkam) [inline release/nightly_tests/dataset/stable_diffusion_benchmark.py:14]: is this the largest batch size that can be used?

**Review by amogkam — APPROVED** [review]

File: release/nightly_tests/dataset/stable_diffusion_benchmark.py (hunk 1 above)
Comment (bveeramani) [reply release/nightly_tests/dataset/stable_diffusion_benchmark.py]: No, but I ran some experiments, and it seems like throughput declines at higher batch sizes. Also, we achieve virtually 100% GPU utilization with lower batch sizes, so I don't think it'll be an issue.

**Comment by bveeramani** [comment]
Release test is passing https://buildkite.com/ray-project/release-tests-pr/builds/53990#018ad88e-21c8-406d-950e-db4ad584fad5

**Review by can-anyscale — APPROVED** [review]
LGTM!

**Pull request merged**